docs/docs/gatsby-style-guide.md

```diff
@@ -278,3 +289,156 @@ The Hemingway App will assign a “grade level” for your writing. You should a
+- bits of source code to study and/or copy
+- step-by-step tutorials for advanced tasks
+- understanding how Gatsby works at a deep level, so deep that they could actually modify or customize their local stuff, or contribute to Gatsby core
+- how does Gatsby really work with Redux, React, and GraphQL?
+- error messages that tell them if it’s a known bug/issue, lead them to docs, and/or suggest fixes
+- guides for how things work in Gatsby
+- they usually already have strong opinions or requirements about what they want to use as their CMS or data source and want to know best practices for their workflow
+- signs that Gatsby is a reliable, long-term choice (signs that it is growing and improving and evidence it will be around for a long time)
+- ways to check their project’s requirements against what Gatsby offers
+- open source code from well-built example sites
+
+### Docs purpose
+
+By referencing the docs, a user should:
+
+- get tasks done as quickly as possible
+- evaluate options for getting tasks done as quickly as possible
+- build sites and apps as quickly as possible, including the following kinds of sites:
+  - marketing
+  - blogs
+  - portfolio
+  - e-commerce
+  - authentication
+
+### Docs tone and style
+
+The tone and style of the docs should effectively help the audience reach their goals.
+
+### Use personal "you"
+
+The docs use the personal “you” to address the user.
+
+#### Give experts as much relevant info to get the task done as quickly as possible
+
+Because the audience of the docs is people who have intermediate to expert level of mastery with React, it's important to provide the information needed to complete tasks in addition to all relevant and helpful context, references,and alternatives. The goal: give people the information necessary to get tasks done as quickly and effectively as possible.
+
+In practice, you can reach this goal by two rules of thumb:
+
+- Include an "additional information" section at the bottom of each guide with hyperlinks to relevant external blogposts, tutorials, and other Gatsby resources and docs.
+- When there are multiple ways to complete a task, explain the following:
+
+1.  The most common way of accomplishing the task
+2.  The best way to accomplish the task on the lowest supported versions of software
+3.  The best practice and why is it the best (if different than 3)
+4.  Any tips on how to pick an option
+
+## KPIs
+
+Here is how we measure the quality of the tutorial and docs.
+
+### Tutorial
+
+- time to value: how quickly, effectively, and painlessly can users experience the value of Gatsby?
+- daily, weekly, monthly active users
+- daily, weekly, monthly “finishers” of the tutorial (finisher = people who spend 5 mins minimum on each page of tutorial and go through whole thing)
+- retention
+
+### Docs
+
+- time to task: how quickly and effectively can users accomplish tasks?
+- daily, weekly, monthly active users
+- retention
```

Comment:
This should be "weekly retention rate" https://www.geckoboard.com/learn/kpi-examples/mobile-app-kpis/retention-rate/

Reply:
for both docs and tutorial KPI?

Reply:
for both docs and tutorial I assume?

Also that site is beautiful!

Reply:
Is this why you say weekly? Bc it sounds like specific time periods is what they recommend, though it seems like the time periods would be oriented around product releases
> Tracking only the aggregate Retention Rate may distort the current trend of retaining users. For example, if you make a significant improvement to the user experience, your retention rate might go up. But if you’re still tracking it in aggregate, you may not notice very much of a change.
> This is why cohort analysis - tracking and comparing Retention Rates by specific time periods - is so important.

Reply:
The length of retention depends on the rhythm of how people use a product/service. So for docs, someone looking through Gatsby and then coming back the next many weeks to keep working on a project sounds like someone that we've "retained". Daily retention sounds too fast of rate because many people work on multiple projects at once and could be very excited about Gatsby and using it regularly but still not visit us daily. Monthly seems too long as if you only visit gatsbyjs.org monthly, you're barely using it.

Reply:
Cohort analysis is necessary to pick out the effect of changes to a product but isn't specific to retention. Cohort analysis should be used with all KPIs so we can track improvements (or regressions) to the docs.